### PR TITLE
Add discord.ChannelType.category to documentation

### PR DESCRIPTION
**Pull request by Denayder**
### Summary

`discord.ChannelType.category` was missing from the documentation.

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -816,6 +816,9 @@ of :class:`enum.Enum`.
     .. attribute:: group
 
         A private group text channel.
+    .. attribute:: category
+
+       A category channel.
```
Comment (Harmon758) [inline docs/api.rst]: This indentation is off:
```suggestion
        A category channel.
```

**Comment by Harmon758** [comment]
Resolves #2558

File: docs/api.rst (hunk 1 above)
Comment (Denayder) [reply docs/api.rst]: Sorry, not sure how I didn't see that 🤦‍♂

**Comment by Rapptz** [comment]
Thanks.

**Pull request closed without merging**